hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto

```diff
@@ -1815,6 +1818,16 @@ message SnapshotPurgeRequest {
   repeated string updatedSnapshotDBKey = 2;
 }
 
+message SetSnapshotPropertyRequest {
+  repeated SnapshotProperty snapshotProperty = 1;
```

Comment:
I initially thought of making it a single API call for each request. But for 1000 snapshots, we will be sending 1000 requests. We are only updating the snapshot metadata, so I see no harm in sending batch requests. But I'm open to change it.

Reply:
@hemantk-12 

For error handling, the caller has to handle any error in the response carefully when some of the requests failed and some succeeded.

Performance wise I'd say we just need to count the metrics in finer granularity. For a `SetSnapshotPropertyRequest` that contains 100 snapshot updates, the metric could be increased by 100 rather than 1.

Reply:
@smengcl 
* My concern is on server side error handling, e.g. in [OMSnapshotSetPropertyRequest](https://github.com/apache/ozone/pull/5301/files#diff-cd87e80aa08edcaf38ee47c0d14820c15419b28e39eed9a0167d4a86db51e76dR87) and [OMSnapshotPurgeRequest](https://github.com/apache/ozone/blob/eec1ce49c87595c83d6946c019c24bda4e86401d/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotPurgeRequest.java#L95) is something fails in the loop, it will be partially updated. It won't be a problem if we handle each entry individually.


* Regarding performance, it is not just about request count metric, that is easier compare to latency and failure metrics. E.g. latency for a batch of 1000 would be different that a batch of 1 or 2. You can argue that we can use avg but it won't give correct picture.

Again, as I said earlier these concerns might not be valid in this case because this API will be used by internal [KeyDeletingService] only.